pkg/roachpb/errors.go

```diff
@@ -1280,3 +1281,31 @@ func (e *OptimisticEvalConflictsError) Type() ErrorDetailType {
 }
 
 var _ ErrorDetailInterface = &OptimisticEvalConflictsError{}
+
+// NewMinTimestampBoundUnsatisfiableError initializes a new
+// MinTimestampBoundUnsatisfiableError.
+func NewMinTimestampBoundUnsatisfiableError(
+	minTimestampBound, resolvedTimestamp hlc.Timestamp,
+) *MinTimestampBoundUnsatisfiableError {
+	return &MinTimestampBoundUnsatisfiableError{
+		MinTimestampBound: minTimestampBound,
+		ResolvedTimestamp: resolvedTimestamp,
+	}
+}
+
+func (e *MinTimestampBoundUnsatisfiableError) Error() string {
+	return e.message(nil)
```

Comment:
should we add a pgcode to this? or is there a place up the stack we usually add a pgcode to this kind of thing?

Reply:
We don't have a way that I'm aware of to attach pgcodes to KV-layer errors. Instead, we have a translation layer in `pkg/sql/row`. See `ConvertBatchError` (which is growing a bit in https://github.com/cockroachdb/cockroach/pull/68042).